utils/docker/vcluster/daos-base/el8/Dockerfile

```
@@ -0,0 +1,92 @@
+
+# Pull base image
+ARG	RHEL_BASE_IMAGE=centos
+ARG	RHEL_BASE_VERSION=8
+FROM	$RHEL_BASE_IMAGE:$RHEL_BASE_VERSION
+LABEL	maintainer="daos@daos.groups.io"
+
+# Configure systemd: more details could be found at following URL:
+# https://markandruth.co.uk/2020/10/10/running-systemd-inside-a-centos-8-docker-container
+# XXX FIXME XXX Should be removed in production with application dedicated entry point
+VOLUME	[ "/sys/fs/cgroup" ]
+RUN	systemctl mask systemd-remount-fs.service graphical.target kdump.service &&                \
+	systemctl unmask dev-hugepages.mount &&                                                    \
+	pushd /lib/systemd/system/sysinit.target.wants &&                                          \
+	for i in * ; do                                                                            \
+		[[ $i == systemd-tmpfiles-setup.service ]] || rm -f $i ;                           \
+	done &&                                                                                    \
+	popd &&                                                                                    \
+	rm -f /lib/systemd/system/multi-user.target.wants/* &&                                     \
+	rm -f /etc/systemd/system/*.wants/* &&                                                     \
+	rm -f /lib/systemd/system/local-fs.target.wants/* &&                                       \
+	rm -f /lib/systemd/system/sockets.target.wants/*udev* &&                                   \
+	rm -f /lib/systemd/system/sockets.target.wants/*initctl* &&                                \
+	rm -f /lib/systemd/system/basic.target.wants/* &&                                          \
+	rm -f /lib/systemd/system/anaconda.target.wants/*
+STOPSIGNAL SIGRTMIN+3
+ENTRYPOINT [ "/sbin/init" ]
+
+# Base configuration of dnf
+RUN	dnf clean all &&                                                                           \
+	dnf makecache &&                                                                           \
+	dnf --assumeyes install dnf-plugins-core &&                                                \
+	dnf config-manager --save --setopt=assumeyes=True &&                                       \
+	dnf config-manager --save --setopt=fastestmirror=True &&                                   \
+	dnf config-manager --set-enabled powertools &&                                             \
+	dnf install epel-release
+
+# Update system
+RUN	dnf update
+
+# XXX NOTE XXX Variale allowing to build the image wihout using glolbal --no-cache option and thus
+# XXX NOTE XXX to not update all rpms.  To work properly a random value such as the date of the day
+# XXX NOTE XXX should be given.
+ARG	BUST_CACHE
+ARG	DAOS_REPOS="https://packages.daos.io/v2.0"
+ARG	DAOS_GPG_KEYS="https://packages.daos.io/RPM-GPG-KEY"
+ARG	DAOS_REPOS_NOAUTH=""
+RUN	[[ $BUST_CACHE ]] && dnf update ;                                                          \
```

You could use -n or -z here to avoid the double braces.  Dockerfiles are odd as each part needs to succeed so I'm not sure which you need to use precisely.
```suggestion
RUN	[ -n "$BUST_CACHE" ] && dnf update ;                                                          \
```

---

The result of this command is not relevant for the docker file as it is ended with ";".
However, having a command evaluating to true is cleaner => I will change the implicit -n by an explicit -z.

In general I prefer the double braces in bash script,  but for dockerfile, I am agree that it is better to use the test command instead of the bash builtins double braces: no reason to be bash scpecific with a Dockerfile.
Thus, I will change all the double braches of the docker files.

---

fixed